front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styled from '@emotion/styled';
+import { useRecoilState } from 'recoil';
+
+import { isNavbarOpenedState } from '@/ui/layout/states/isNavbarOpenedState';
+import { isRightDrawerExpandedState } from '@/ui/right-drawer/states/isRightDrawerExpandedState';
+import { useListenClickOutside } from '@/ui/utilities/pointer-event/hooks/useListenClickOutside';
+
+import { Chip, ChipVariant } from '../chip/components/Chip';
+
+type OwnProps = {
+  isOverflowOpen: boolean;
+  width: number;
+};
+
+const StyledContainer = styled.div<Partial<OwnProps>>`
+  display: flex;
+  flex: 1;
+  gap: ${({ theme }) => theme.spacing(1)};
+  justify-content: flex-end;
+  max-width: calc(100vw - 265px);
+  overflow: hidden;
+  padding: ${({ isOverflowOpen, theme }) =>
+    isOverflowOpen ? theme.spacing(4) : theme.spacing(1)};
+  width: ${({ width }) => (width === 0 ? '100%' : `${width}px`)};
+`;
+
+const StyledOverflowContainer = styled.div`
+  background-color: ${({ theme }) => theme.background.secondary};
+  border-radius: ${({ theme }) => theme.border.radius.sm};
+  box-shadow: ${({ theme }) => theme.boxShadow.strong};
+  display: flex;
+  flex-wrap: wrap;
+  gap: ${({ theme }) => theme.spacing(2)};
+  padding: ${({ theme }) => theme.spacing(2)};
+  width: 100%;
+  z-index: 9999;
+`;
+
+export function OverflowingChipsWithTooltip({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const [isOverflowOpen, setIsOverflowOpen] = useState(false);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [visibleItems, setVisibleItems] = useState<React.ReactNode[]>([]);
+  const [remaningItems, setRemaningItems] = useState(0);
+
+  const [isNavOpen] = useRecoilState(isNavbarOpenedState);
+  const [isRightDrawerExpanded] = useRecoilState(isRightDrawerExpandedState);
+
+  useEffect(() => {
+    const element = containerRef?.current;
+    if (!element) return;
+
+    const resizeObserver = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.target === containerRef.current) {
+          setContainerWidth(entry.contentRect.width);
+        }
+      });
+    });
+
+    resizeObserver.observe(containerRef.current);
+
+    return () => {
```

Comment:
Why is this return statement required ? Nit: commented code

Reply:
It is recommended to clear the side effects during unmounting of a component, that's why I wrote this part. I will clear the comment.